pkg/controller/migration/convert/network.go

```diff
@@ -119,6 +126,107 @@ func handleNetwork(c *components, install *Installation) error {
+	if err != nil {
+		return err
+	}
+
+	if install.Spec.CNI == nil {
+		install.Spec.CNI = &operatorv1.CNISpec{}
+	}
+
+	if prefix == nil {
+		return ErrIncompatibleCluster{"with no CNI config, expected FELIX_INTERFACEPREFIX to be set"}
+	}
+	switch *prefix {
+	case "eni":
+		install.Spec.CNI.Type = operatorv1.PluginAmazonVPC
+		// For AmazonVPC, check that IPTABLESMANGLE is set Return
+		if mangleAllow == nil {
+			return ErrIncompatibleCluster{"detected AmazonVPC CNI plugin, FELIX_IPTABLESMANGLEALLOWACTION was unset, expected it to be 'Return'."}
+		} else if *mangleAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf("detected AmazonVPC CNI plugin, FELIX_IPTABLESMANGLEALLOWACTION was %s, expected it to be 'Return'.", *mangleAllow)}
+		}
+		// For AmazonVPC, check that IPTABLESFILTER is not set or the default Accept
+		if filterAllow != nil && *filterAllow != "Accept" {
+			return ErrIncompatibleCluster{fmt.Sprintf(
+				"detected AmazonVPC CNI plugin, FELIX_IPTABLESFILTERALLOWACTION was set to %s, expected it to be unset or 'Accept'.",
+				*filterAllow)}
+		}
+	case "avz":
+		install.Spec.CNI.Type = operatorv1.PluginAzureVNET
+		// For AzureVNET, check that IPTABLESMANGLE is not set or the default Accept
+		if mangleAllow != nil && *mangleAllow != "Accept" {
+			return ErrIncompatibleCluster{fmt.Sprintf(
+				"detected AzureVNET CNI plugin, FELIX_IPTABLESMANGLEALLOWACTION was set to %s, expected it to be unset or 'Accept'.",
+				*mangleAllow)}
+		}
+		// For AzureVNET, check that IPTABLESFILTER is not set or the default Accept
+		if filterAllow != nil && *filterAllow != "Accept" {
+			return ErrIncompatibleCluster{fmt.Sprintf(
+				"detected AzureVNET CNI plugin, FELIX_IPTABLESFILTERALLOWACTION was set to %s, expected it to be unset or 'Accept'.",
+				*filterAllow)}
+		}
+	case "gke":
+		install.Spec.CNI.Type = operatorv1.PluginGKE
+		// For GKE, check that IPTABLESMANGLE is set Return
+		if mangleAllow == nil {
+			return ErrIncompatibleCluster{"detected GKE CNI plugin, FELIX_IPTABLESMANGLEALLOWACTION was unset, expected it to be 'Return'."}
+		} else if *mangleAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf("detected GKE CNI plugin, FELIX_IPTABLESMANGLEALLOWACTION was %s, expected it to be 'Return'.", *mangleAllow)}
+		}
+		// For GKE, check that IPTABLESFILTER is set Return
+		if filterAllow == nil {
+			return ErrIncompatibleCluster{"detected GKE CNI plugin, FELIX_IPTABLESFILTERALLOWACTION was unset, expected it to be 'Return'."}
+		} else if *filterAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf(
+				"detected GKE CNI plugin, FELIX_IPTABLESFILTERALLOWACTION was set to %s, expected it to be 'Return'.",
+				*filterAllow)}
+		}
+	default:
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected FELIX_INTERFACEPREFIX value: '%s'. Only 'eni, avz, gke' are supported.", *prefix),
+		}
+	}
+
+	c.node.ignoreEnv(containerCalicoNode, "IP")
```

Comment:
is this `TODO` or permanent?

Reply:
It is until we support non-Calico CNI and CalicoNetwork.

Reply:
Actually no, ignore probably isn't right.